akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -292,6 +292,7 @@ class TLSActor(
       if (tracing) log.debug("closing inbound")
       try engine.closeInbound()
       catch { case ex: SSLException ⇒ outputBunch.enqueue(UserOut, SessionTruncated) }
+      lastHandshakeStatus = engine.getHandshakeStatus
```

Comment:
Thanks for the explanation. Can you put that into the commit message as well?

I guess one could also say that whenever you signal something to the SSLEngine, you should also read getHandshakeStatus afterwards to understand what you need to do next. This is done everywhere but for `engine.closeInbound`.

Can you also add a comment to the `engine.closeInbound` a few lines below, why we don't need to add this there as well? (IIUC it's because we are there in a case where it doesn't make any sense to write anything to the network anymore).